tree/tree/src/TTree.cxx

```diff
@@ -4384,13 +4384,24 @@ Int_t TTree::Fill()
    if (fBranchRef) {
       fBranchRef->Clear();
    }
+
+   std::unique_ptr<TBranchIMTHelper> imt_helper;
```

Comment:
To avoid any overheads when IMT is disabled - or the `R__USE_IMT` is not defined - the `FillImpl` function expects a `nullptr`.

Hence, this avoids doing this repeatedly:

```
branch->FillImpl(doImt ? &imt_helper : nullptr);
```

Hm - I suppose we could use the uglier syntax and avoid a heap allocation though.  What's your preference?